cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/ApplyLogic.java

```diff
@@ -84,227 +86,127 @@ public void setSpaceOperations(SpaceOperations spaceOperations) {
-     * non-recoverable error, the procedure is discontinued.
+     * Provides the service of manipulating the state of a cloud foundry instance
+     * such that it matches with a desired configuration ({@link ConfigBean}).
      *
-     * @param desiredApplications the applications that should all be present in the
-     *                            live system after the procedure
-     * @throws ApplyException       if an non-recoverable error occurs during the
-     *                              procedure
-     * @throws NullPointerException if the argument is null
+     * @param desiredConfigBean   desired configuration for a cloud foundry instance
+     * @param loginCommandOptions LoginCommandOptions
+     * @throws NullPointerException if one of the desired parameters is null.
      */
-    public void applyApplications(@Nonnull Map<String, ApplicationBean> desiredApplications) {
-        checkNotNull(desiredApplications);
+    public void apply(ConfigBean desiredConfigBean, OptionalLoginCommandOptions loginCommandOptions) {
+        checkNotNull(desiredConfigBean);
+        checkNotNull(loginCommandOptions);
+        checkNotNull(desiredConfigBean.getTarget(), "Target bean may not be null.");
+        checkNotNull(desiredConfigBean.getTarget().getSpace(), "Space may not be null.");
 
-        log.info("Fetching applications data");
-        Map<String, ApplicationBean> liveApplications = this.getLogic.getApplications(applicationsOperations);
-        log.verbose("Fetched space developers data");
-
-        // that way only the applications of the live system are compared in the diff
-        ConfigBean desiredApplicationsConfig = createConfigFromApplications(desiredApplications);
-        ConfigBean liveApplicationsConfig = createConfigFromApplications(liveApplications);
+        try {
+            Mono<List<String>> getAllRequest = spaceOperations.getAll();
 
-        // compare entire configs as the diff wrapper is only suited for diff trees of
-        // these
-        log.info("Diffing applications");
+            log.info("Fetching names of all spaces");
+            List<String> spaceNames = getAllRequest.block();
+            log.verbose("Fetching names of all spaces completed");
 
-        DiffResult diffResult = this.diffLogic.createDiffResult(liveApplicationsConfig, desiredApplicationsConfig);
-        log.verbose("Diffing applications completed");
+            // getting
 
-        Map<String, List<CfChange>> allApplicationChanges = diffResult.getApplicationChanges();
+            ConfigBean liveConfigBean = new ConfigBean();
 
-        if (allApplicationChanges == null || allApplicationChanges.isEmpty()) {
-            log.info("No changes detected, nothing to apply");
-        } else {
-            ApplicationRequestsPlanner requestsPlanner = new ApplicationRequestsPlanner(applicationsOperations);
+            String desiredSpaceName = desiredConfigBean.getTarget().getSpace();
```

Comment:
As the desired space might also have been in the `LoginCommandOptions`:
```suggestion
            String desiredSpaceName = targetOperations.getSpace();
```